bundles/binding/org.openhab.binding.powerdoglocalapi/src/main/java/org/openhab/binding/powerdoglocalapi/internal/PowerDogLocalApiBinding.java

```diff
@@ -0,0 +1,572 @@
+			}
+			
+			setProperlyConfigured(true);
+			logger.debug("PowerDogLocalApi:parseConfiguration() method is terminated");	
+		}
+	}
+	
+	static class PowerDogLocalApiServerConfig {
+		public String host;		// IP adress or DNS entry
+		public int port;		// port number
+		public String password;	// password
+		public int refresh;		// refresh rate in ms
+		public Long lastUpdate;	// saves last update time when xmlrpc was read
+		public XmlRpcStruct cache;
+
+		PowerDogLocalApiServerConfig() {
+			lastUpdate = (long) 0;
+			
+			// set defaults
+			refresh = 300000; 	// 5 min is default
+			password = "";		// empty password will normally not be accepted by PowerDog, needs to be configured
+			port = 20000; 		// port 20000 is default for PowerDog
+			host = "powerdog";	// local DNS in router might resolve this one
+		}
+		
+		@Override
+		public String toString() {
+			String displayPassword = "[not set]";
+			if(StringUtils.isNotBlank(password)) {
+				displayPassword = "[set]*****";
+			}
+			return "PowerDogLocalApiServerCache [host="+host+", password=" + displayPassword + ", lastUpdate=" + lastUpdate + ", cache=" + cache + "]";
+		}
+		
+		
+		public URL url() throws MalformedURLException {
+			return new URL("http", host, port, "");
+		}
+
+	}
+	
+	/**
+	 * PowerAPI Local Device API 0.b (15.02.2013)
+	 * Tested against PowerDog Firmware V1.84 (High Velocity) 2013-09-19
+	 * 
+	 * VariantMap getPowerDogInfo(String password);
+	 * VariantMap getSensors(String password);
+	 * VariantMap getCounters(String password);
+	 * VariantMap getRegulations(String password);
+	 * VariantMap getLinearDevices(String password);
+	 * VariantMap getAllCurrentLinearValues(String password);
+	 * VariantMap getCurrentLinearValues(String password, String comma_seperated_list_of_keys);
+     * VariantMap setLinearSensorDevice(String password, String key, String current_value);
+     * VariantMap setLinearCounterDevice(String password, String key, String current_value, String countup_meter_reading);
+	 * 
+	 * @author Wuellueb
+	 *
+	 */
+	static interface PowerDog
+	{
+	    public XmlRpcStruct getPowerDogInfo( String password ) throws XmlRpcFault;
+	    public XmlRpcStruct getSensors( String password ) throws XmlRpcFault;
+	    public XmlRpcStruct getCounters( String password ) throws XmlRpcFault;
+	    public XmlRpcStruct getRegulations( String password ) throws XmlRpcFault;
+	    public XmlRpcStruct getLinearDevices( String password ) throws XmlRpcFault;
+	    public XmlRpcStruct getAllCurrentLinearValues( String password ) throws XmlRpcFault;
+	    public XmlRpcStruct getCurrentLinearValues( String password, String comma_seperated_list_of_keys ) throws XmlRpcFault;
+	    public XmlRpcStruct setLinearSensorDevice( String password, String key, String current_value ) throws XmlRpcFault;
+	    public XmlRpcStruct setLinearCounterDevice( String password, String key, String current_value, String countup_meter_reading ) throws XmlRpcFault;
+	}
```

Comment:
I added some comments to the interface and referenced the XML RPC interface contract documentation of the PowerDog producer. I left the interface inline because it is used within this class only to cast the XmlRpcProxy accordingly and because there may be nowhere an actual implementation of this interface.